### ISSUE TITLE
Extents that are not 'recorded and allocated' not implemented

### ISSUE THREAD

**Issue by momaqbol** [Finzenku/Ps2IsoTools#5]
I keep having the following exception and I'm not sure if it's because of the source .iso file or something else, I'm running it using dotnet v8.0.1 linux.
```
Unhandled exception. System.NotImplementedException: Extents that are not 'recorded and allocated' not implemented
   at Ps2IsoTools.DiscUtils.Streams.Buffers.FileContentBuffer.LoadExtents()
   at Ps2IsoTools.DiscUtils.Streams.Buffers.FileContentBuffer..ctor(UdfContext context, Partition partition, FileEntry fileEntry, UInt32 blockSize)
   at Ps2IsoTools.UDF.Files.File..ctor(UdfContext context, Partition partition, FileEntry fileEntry, UInt32 blockSize)
   at Ps2IsoTools.UDF.Files.File.FromDescriptor(UdfContext context, LongAllocationDescriptor icb)
   at Ps2IsoTools.UDF.UdfReader.ConvertFileIdToFile(FileIdentifier dirEntry)
   at Ps2IsoTools.UDF.UdfReader.GetFileStream(FileIdentifier file)
   at Ps2IsoTools.UDF.UdfEditor.InitializeReader(String fileToRead)
   at Ps2IsoTools.UDF.UdfEditor..ctor(String inputFile, String copyTo)
   at Program.<Main>$(String[] args) in /home/MyUsername/Desktop/TheWarriors/ps2tools/Program.cs:line 5
```
I used the same "Editing a UDF format ISO with UdfEditor" example in the readme.

**Comment by Finzenku** [comment]
The `FileContentBuffer` is a file taken directly from [DiscUtils](https://github.com/DiscUtils/DiscUtils) so I'm not super well versed in the specifics of it.
The `ShortAllocationDescriptor` that the `FileContentBuffer` uses to load each extent has a set of flags that seem to almost always be set as "recorded and allocated". It looks like they decided to only implement that specific case.

Looking at the specification documentation, the flag would only be different in a few cases like if the file wasn't actually written on the disc or if the file was scattered across multiple places.

What game are you trying to work with? It might help if I could look at the file structure and see what case we're dealing with.

**Comment by momaqbol** [comment]
I'm trying to work with The Warriors (USA), SLUS_212.15.

I'm going to try out other games to see if it gets fixed

**Comment by 0za0** [comment]
I am also encountering this issue with Bionicle Heroes SLUS-21428, is there any way to bypass this and still extract/edit the iso contents?

**Comment by Finzenku** [comment]
After looking at both The Warriors and Bionicle Heroes, the issue seems to be specifically with files that are larger than 1GB.
The `ShortAllocationDescriptor` flag on those files is set to "allocated not recorded" which would suggest that the files weren't written to the disc even though they clearly are. I can only assume that the PS2 doesn't use this flag in favor of having large files with a single descriptor. 

This commit seems to fix the issue, I was able to read and rebuild both games and run then in PCSX2. I've submitted a new version of the nuget package (v1.1.1) with the changes. Let me know if any other issues come up.
